lib/Logger/LogAppenderFile.cpp

```diff
@@ -55,48 +54,21 @@ int LogAppenderFile::_fileMode = S_IRUSR | S_IWUSR | S_IRGRP;
 int LogAppenderFile::_fileGroup = 0;
 
 LogAppenderStream::LogAppenderStream(std::string const& filename, int fd)
-    : LogAppender(),
-      _bufferSize(0),
-      _fd(fd),
-      _useColors(false),
-      _controlEscape(Logger::getUseControlEscaped()),
-      _unicodeEscape(Logger::getUseUnicodeEscaped()) {
-  if (_controlEscape) {
-    if (_unicodeEscape) {
-      _escaper =
-          std::make_unique<Escaper<ControlCharsEscaper, UnicodeCharsEscaper>>();
-    } else {
-      _escaper = std::make_unique<
-          Escaper<ControlCharsEscaper, UnicodeCharsRetainer>>();
-    }
-  } else {
-    if (_unicodeEscape) {
-      _escaper = std::make_unique<
-          Escaper<ControlCharsSuppressor, UnicodeCharsEscaper>>();
-    } else {
-      _escaper = std::make_unique<
-          Escaper<ControlCharsSuppressor, UnicodeCharsRetainer>>();
-    }
-  }
-}
-
-size_t LogAppenderStream::determineOutputBufferSize(
-    std::string const& message) const {
-  return _escaper->determineOutputBufferSize(message) +
-         2;  //+2 bytes because it needs to end with '\n' and '\0'
-}
+    : LogAppender(), _bufferSize(0), _fd(fd), _useColors(false) {}
 
 size_t LogAppenderStream::writeIntoOutputBuffer(std::string const& message) {
+  std::string formattedMsg = message + '\n';
   char* output = _buffer.get();
-  _escaper->writeIntoOutputBuffer(message, output);
-  *output++ = '\n';
-  *output = '\0';
-  return (output - _buffer.get());
+
+  strncpy(output, formattedMsg.data(), formattedMsg.size());
+  output[formattedMsg.size()] = '\0';
+
+  return (output + strlen(output) - _buffer.get());
```

Comment:
This will create a new std::string for each message, plus a new temporary one, and then call `strlen()` in addition. That looks like horrible performance. This is a critical part of code that needs to be a fast as possible.